File: server/src/game/tokenCards.ts
import { CardType, Rarity, HeroClass, type CardDefinition } from '@coincard/shared';

// Token sinh trong trận (hero power / Polymorph) — không có trong deck.
export const SILVER_HAND_RECRUIT_TOKEN: CardDefinition = {
  id: 'token_silver-hand-recruit',
  name: 'Silver Hand Recruit',
  slug: 'silver-hand-recruit',
  description: 'Hero Power token.',
  type: CardType.MINION,
  rarity: Rarity.COMMON,
  manaCost: 1,
  attack: 1,
  health: 1,
  heroClass: HeroClass.PALADIN,
  imagePath: 'assets/images/Minions/Silver Hand Recruit.png',
  effects: [],
  keywords: [],
  collectible: false,
};

export const SHEEP_TOKEN: CardDefinition = {
  id: 'token_sheep',
  name: 'Sheep',
  slug: 'sheep',
  description: 'Polymorph token.',
  type: CardType.MINION,
  rarity: Rarity.COMMON,
  manaCost: 1,
  attack: 1,
  health: 1,
  heroClass: HeroClass.MAGE,
  imagePath: 'assets/images/Minions/Sheep.png',
  effects: [],
  keywords: [],
  collectible: false,
};

File: server-java/src/main/java/vn/coincard/server/game/TokenCards.java
package vn.coincard.server.game;

import java.util.List;

/** Non-collectible cards created by effects and hero powers. */
public final class TokenCards {
  private TokenCards() {}

  public static final CardTypes.CardDefinition SILVER_HAND_RECRUIT_TOKEN =
      new CardTypes.CardDefinition(
          "token-silver-hand", "Silver Hand Recruit", "silver-hand-recruit",
          "Hero Power token.", CardType.MINION, Rarity.COMMON, 1, 1, 1, HeroClass.PALADIN,
          "assets/images/Minions/Silver Hand Recruit.png",
          List.of(), List.of(), false);

  public static final CardTypes.CardDefinition SHEEP_TOKEN =
      new CardTypes.CardDefinition(
          "token-sheep", "Sheep", "sheep",
          "Polymorph token.", CardType.MINION, Rarity.COMMON, 1, 1, 1, HeroClass.MAGE,
          "assets/images/Minions/Sheep.png",
      List.of(), List.of(), false);

  public static CardTypes.CardDefinition coinCard() {
    return new CardTypes.CardDefinition(
        "coin-token-" + java.util.UUID.randomUUID(), "The Coin", "the-coin",
        "Gain 1 temporary Mana Crystal this turn.", CardType.SPELL, Rarity.COMMON, 0, 0, 0,
        HeroClass.NEUTRAL, "assets/images/design/manacrystal.png",
        List.of(new CardTypes.EffectDefinition(
            EffectType.TEMPORARY_MANA, 1, EffectTarget.SELF, null, null, null)),
        List.of(), false);
  }
}
